backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/material/cockpit/MaterialCockpitRowCache.java

```diff
@@ -22,8 +27,9 @@
 @RequiredArgsConstructor
 public class MaterialCockpitRowCache
 {
-	private final ProductRepository productRepository;
-	private final IUOMDAO uomDAO;
+	@NonNull private final ProductRepository productRepository = SpringContextHolder.instance.getBean(ProductRepository.class);
```

Comment:
pls avoid getBean here. Provide those as ctor params.

Reply:
Then I run into may not be initialized error and I couldn't figure out why